docs/source/API/containers/Array.rst

```diff
@@ -0,0 +1,173 @@
+
+..
+  Class / method / container name)
+  for free functions that are callable, preserve the naming convention, `view_alloc()`
+
+``Array``
+==============
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+..
+  The (pulic header) file the user will include in their code
+
+Header File: ``Kokkos_Core.hpp``
+
+..
+  High-level, human-language summary of what the thing does, and if possible, brief statement about why it exists (2 - 3 sentences, max);
+
+Description
+-----------
+
+``Array`` is a contiguous aggregate owning container storing a fixed size sequence of objects (models holding exactly N elements).
+
+* This container is an owning container (the data is embeddded in the container itself).
+* This container is an aggregate type with the same semantics as a struct holding a C-style array ``T[N]`` as its only non-static data member when ``N > 0``; otherwise, it is an empty container.
+* Unlike a C-style array, it doesn't decay to ``T*`` automatically.
+* As an aggregate type, it can be initialized with aggregate-initialization given at most ``N`` initializers that are convertible to ``T``: ``Kokkos::Array<int, 3> a = { 1, 2, 3 };``.
+
+..
+  The API of the entity.
+
+Interface
+---------
+
+..
+  The declaration or signature of the entity.
+
+.. cppkokkos:struct:: template <class T, size_t N> Array
+
+  ..
+    Template parameters (if applicable)
+    Omit template parameters that are just used for specialization/are deduced/ and/or should not be exposed to the user.
+
+  .. rubric:: Template Parameters
+
+  :tparam T: The type of the element being stored.
+  :tparam N: The number of elements being stored.
+
+  .. rubric:: Public Types
+
+  .. cppkokkos:type:: value_type = T
+  .. cppkokkos:type:: pointer = T*
+  .. cppkokkos:type:: const_pointer = const T*
+  .. cppkokkos:type:: reference = T&
+  .. cppkokkos:type:: const_reference = const T&
+  .. cppkokkos:type:: size_type = size_t
+  .. cppkokkos:type:: difference_type = ptrdiff_t
+
+  .. rubric:: Public Member Functions
+
+  .. cppkokkos:function:: static constexpr bool empty()
+
+    :return: ``N == 0``
+
+  .. cppkokkos:function:: static constexpr size_type size()
+  .. cppkokkos:function:: constexpr size_type max_size() const
+
+    :return: ``N``
+
+  .. cppkokkos:function:: template<class iType> constexpr reference operator[](const iType& i)
+  .. cppkokkos:function:: template<class iType> constexpr const_reference operator[](const iType& i) const
+
+    :tparam iType: An integral type or an unscoped enum type.
+
+    :return: A reference to the ``i``-th element of the array.
+
+  .. cppkokkos:function:: constexpr pointer data()
+  .. cppkokkos:function:: constexpr const_pointer data() const
+
+    :return: A pointer to the first element of the array.  If ``N == 0``, the return value is unspecified and not dereferenceable.
+
+
+Deduction Guides
+----------------
+
+.. cppkokkos:function:: template<class T, class... U> Array(T, U...) -> Array<T, 1 + sizeof...(U)>
```

Comment:
maybe this can be highlighted in an example

Reply:
There is already one in the Examples section.  Should we have more than that?